relay-general/src/types/annotated.rs

```diff
@@ -253,21 +232,33 @@ where
     pub fn deserialize_with_meta<'de, D: Deserializer<'de>>(
         deserializer: D,
     ) -> Result<Self, D::Error> {
-        Ok(FromValue::from_value(
-            match Option::<Value>::deserialize(deserializer)? {
-                Some(Value::Object(mut map)) => {
-                    let meta_tree = map
-                        .remove("_meta")
-                        .map(MetaTree::from_value)
-                        .unwrap_or_default();
-
-                    let mut value: Annotated<Value> = Annotated::new(Value::Object(map));
-                    value.attach_meta_tree(meta_tree);
-                    value
-                }
-                other => Annotated::from(other),
+        #[derive(Deserialize)]
+        #[serde(untagged)]
```

Comment:
@mitsuhiko @jan-auer curious if there's a more efficient way to write this. I'd hope that serde(untagged) is at least doing slightly more efficient buffering than putting everything in a `Value`.

Reply:
I think can write a deserializer wrapper (impl `Deserializer` on `&mut` of that type) that continues to consume the current level as ignored when it encounters an error.